### PR TITLE
feat(pr-loop-lib): require two consecutive quiescent iterations

### PR DESCRIPTION
**Pull request by prpande**
## Summary

Make the PR-comment loop's quiescence test more robust against late-arriving reviewer-bot findings (Copilot code review, SonarCloud, etc.). The loop now requires **two consecutive quiescent iterations** before exiting, and the per-iteration wait floor drops from 10 minutes to **5 minutes** — the same ~10-minute wall-clock protection, but now spread across two waits with a fresh fetch between them. A late-bot finding that lands in the gap between the two iterations gets picked up by iteration N+1 instead of being silently missed by a single-fetch exit. Design + plan documents are added under `docs/superpowers/`; the change is propagated through `pr-loop-lib`'s context schema, log-format, wait-cycle, and quiescence-check steps, and the `pr-autopilot` / `pr-followup` orchestrator SKILLs.

## Files Changed

| File | Change | Summary |
|------|--------|---------|
| `docs/superpowers/specs/2026-05-06-pr-loop-quiescence-confirmation-design.md` | Added | Design spec: rationale, `loop_exit_reason` enum rewrite, files-touched plan, edge cases, 7 test scenarios |
| `docs/superpowers/plans/2026-05-06-pr-loop-quiescence-confirmation-implementation.md` | Added | 7-task implementation plan with grep-based cross-file consistency sweep |
| `skills/pr-tooling/pr-loop-lib/references/context-schema.md` | Modified | Added `consecutive_quiescent_iterations` and `last_quiescence_reason` fields; rewrote `loop_exit_reason` enum (3 values); updated `wait_override_minutes` row to reference 5-min floor |
| `skills/pr-tooling/pr-loop-lib/references/log-format.md` | Modified | Added `quiescence_pending` event row; updated `wait_clamped` payload (effective_minutes 10 → 5) |
| `skills/pr-tooling/pr-loop-lib/references/invariants.md` | Modified | Rewrote S08.1 to enumerate the 3-value enum explicitly; replaced S08.3's `{quiescent-*}` wildcard with literal `quiescent-confirmed` (preflight fix) |
| `skills/pr-tooling/pr-loop-lib/steps/01-wait-cycle.md` | Modified | Floor and default 600s → 300s; rewrote "Minimum wait" prose for two-leg protection; cache-note kept (300s ≈ TTL boundary, paired-leg total still past it) |
| `skills/pr-tooling/pr-loop-lib/steps/08-quiescence-check.md` | Modified | New confirmation-counter rule, new routing pseudocode, new `quiescent-confirmed` exit reason; preflight-fix additions: explicit CI re-entry counter reset and clarification that the `quiescence` event fires on every loop exit |
| `skills/pr-tooling/pr-followup/SKILL.md` | Modified | `--wait` flag floor 10 → 5; orchestrator hard-rule reference updated; mention of confirmation iteration so users understand why a quiet first iteration doesn't immediately exit |
| `skills/pr-tooling/pr-autopilot/SKILL.md` | Modified | Same updates as pr-followup; all `10 minute` / `600` callouts re-evaluated per the spec's grep directive |
| `skills/pr-tooling/pr-autopilot/steps/04-open-pr.md` | Modified | Post-open announce string: "Next fetch in 10 minutes" → "Next fetch in 5 minutes (loop exits after two consecutive quiescent iterations)" |

## Security Impact

- [x] No security impact

The change is internal loop-control logic (when to exit a polling loop) and documentation. No auth/authz, secret-handling, input-validation, or external-API surface is touched.

## Testing

- `python scripts/validate.py` → exit 0 (markdown reference graph intact across the touched files; no broken `steps/`, `references/`, or skill-internal includes).
- Preflight adversarial review (Sonnet) caught 6 cross-artifact issues — all fixed inline before this PR opened. See the local review-summary at `.pr-autopilot/branch-pratyush%2Fpr-loop-quiescence-confirmation-review-summary.md` for details.
- The 7 test scenarios in the design spec are paper-traceable end-to-end through the modified prose; that is the bar the design + plan require for a docs-only change to a spec-driven skill.
- End-to-end exercise will happen on this very PR — the quiescence-confirmation behavior gets a real run as the comment-loop processes any reviewer-bot output that arrives.

## Related Work

- Spec: `docs/superpowers/specs/2026-05-06-pr-loop-quiescence-confirmation-design.md`
- Plan: `docs/superpowers/plans/2026-05-06-pr-loop-quiescence-confirmation-implementation.md`

### CODE REVIEW

File: skills/pr-tooling/pr-loop-lib/references/invariants.md [hunk 1]
```diff
@@ -110,9 +110,9 @@ equivalent predicates live in `P02.*`.
 
 | # | Predicate |
 |---|---|
-| S08.1 | `loop_exit_reason` is set to exactly one of the 4 enum values |
+| S08.1 | `loop_exit_reason` is set to exactly one of the 3 enum values: `quiescent-confirmed`, `iteration-cap`, `runaway-detected` (per the `loop_exit_reason` rewrite in `2026-05-06-pr-loop-quiescence-confirmation-design.md`; the soft-quiescent values `quiescent-zero-actionable` and `quiescent-no-code-change` no longer settle on `loop_exit_reason` — they are recorded on `last_quiescence_reason` and the loop continues to the confirmation iteration) |
```
[the remark this answers is not in the document]
Comment (prpande) [reply skills/pr-tooling/pr-loop-lib/references/invariants.md]: > S08.1 currently requires `loop_exit_reason` to be set to one of the 3 enum values at the end of step 08, but the new step-08 flow explicitly leaves `loop_exit_reason` unset on the first soft-quiescent

Addressed: Good catch — S08.1 was unconditional and would fire on the pending-confirmation iteration. Fixed in `c45264b`+follow-up: the invariant now explicitly fires only on actual loop exit (route to step 09 or step 11), and the prose calls out that on a first soft-quiescent iteration `loop_exit_reason` stays unset by design. See the updated S08.1 block.

File: skills/pr-tooling/pr-loop-lib/references/log-format.md [hunk 2]
```diff
@@ -64,6 +64,7 @@ Rules:
 | `commit_pushed` | `{sha, files, message_first_line}` |
 | `reply_posted` | `{feedback_id, thread_id?, surface, resolved}` |
 | `quiescence` | `{reason, loop_exit_reason, termination_reason}` |
+| `quiescence_pending` | `{iteration, reason, next_wait_seconds}` — emitted by `pr-loop-lib/steps/08-quiescence-check.md` when the *first* of two consecutive soft-quiescent iterations completes. `reason` is one of `quiescent-zero-actionable` \| `quiescent-no-code-change`. `next_wait_seconds` is the wait length step 01 will use before the confirming iteration (300 by default; `context.wait_override_minutes * 60` if set). Informational; the loop routes back to step 01 instead of step 09. The next iteration that actually exits (whether confirmed quiescent or by cap/runaway) emits the existing `quiescence` event. |
```
[the remark this answers is not in the document]
Comment (prpande) [reply skills/pr-tooling/pr-loop-lib/references/log-format.md]: > `quiescence_pending.next_wait_seconds` is described as `context.wait_override_minutes * 60` when set, but step 01 clamps via `max(300, wait_override_minutes*60)`. If `--wait` is below the floor (e.g.,

Addressed: Right — the doc said `context.wait_override_minutes * 60` but step 01 actually clamps via `max(300, ...)`. Updated the `quiescence_pending` row to read `max(300, context.wait_override_minutes * 60)` and added an explicit note that sub-floor `--wait` values (e.g., `--wait 2`) are clamped to 300 and reported as 300, not the raw input.

File: skills/pr-tooling/pr-loop-lib/steps/08-quiescence-check.md [hunk 3]
```diff
@@ -43,29 +75,80 @@ classify) triggers a re-entry, the iteration counter resets to 1. The CI
...
+
+elif exit_reason in {quiescent-zero-actionable, quiescent-no-code-change}:
+    context.consecutive_quiescent_iterations += 1
+    context.last_quiescence_reason = exit_reason
+    if context.consecutive_quiescent_iterations >= 2:
+        context.loop_exit_reason = "quiescent-confirmed"
+        emit quiescence event {reason: exit_reason, loop_exit_reason: "quiescent-confirmed", termination_reason: null}
+        proceed to step 09 (CI gate)
+    else:
+        emit quiescence_pending event {iteration: context.iteration, reason: exit_reason, next_wait_seconds: <300 or context.wait_override_minutes*60>}
```
[the remark this answers is not in the document]
Comment (prpande) [reply skills/pr-tooling/pr-loop-lib/steps/08-quiescence-check.md]: > In the routing pseudocode, `next_wait_seconds: <300 or context.wait_override_minutes*60>` doesn’t match step 01’s actual delay computation (`max(300, wait_override_minutes * 60)`) and will be wrong wh

Addressed: Same root cause as the log-format comment — the routing pseudocode was `<300 or context.wait_override_minutes*60>`, which doesn't express the clamp. Updated to `max(300, (context.wait_override_minutes or 5) * 60)` with an inline comment cross-referencing step 01 so the two stay in lock-step.

File: skills/pr-tooling/pr-loop-lib/steps/01-wait-cycle.md [hunk 4]
```diff
@@ -100,28 +115,35 @@ Neither can be inferred from repo inspection.
...
 
 **Required procedure:**
 1. Record the start timestamp (UTC) before sleeping.
-2. Sleep (or suspend via `ScheduleWakeup`) until at least 600 s have
+2. Sleep (or suspend via `ScheduleWakeup`) until at least 300 s have
    elapsed.
-3. Verify elapsed time >= 600 s before invoking step 02.
+3. Verify elapsed time >= 300 s before invoking step 02.
 4. Log a `wait_cycle` event with
    `{iteration, start_ts, end_ts, elapsed_seconds}`.
```
[the remark this answers is not in the document]
Comment (prpande) [reply skills/pr-tooling/pr-loop-lib/steps/01-wait-cycle.md:150]: > The “Manual orchestration” required procedure hard-codes sleeping/validating only `>= 300s`. If `context.wait_override_minutes` is set above the floor (e.g., `--wait 60`), the manual path should still

Addressed: Agreed — manual orchestration was hard-coded to `>= 300s`, which would silently ignore `--wait 60`. Rewrote the "Required procedure" to compute the same `delay_seconds` the ScheduleWakeup path uses (`max(300, (context.wait_override_minutes or 5) * 60)`) and use that value for both the sleep and the elapsed-time gate. Manual and automated paths now have identical wait semantics.
